classes/context/SubEditorsDAO.php

```diff
@@ -113,32 +119,22 @@ public function deleteBySubmissionGroupId($assocId, $assocType, $contextId = nul
 
     /**
      * Delete all submission group assignments for the specified user.
-     *
-     * @param int $userId
-     * @param int $contextId optional, include assignments only in this context
-     * @param int $assocId optional, include only this submission group
-     * @param int $assocType optional ASSOC_TYPE_SECTION or ASSOC_TYPE_CATEGORY
      */
-    public function deleteByUserId($userId, $contextId = null, $assocId = null, $assocType = null)
+    public function deleteByUserId(int $userId)
```

Comment:
If this were not such a new and niche function I would be tempted to add a check at the top:
```php
// TEMPORARY: This check was added with 3.4.0 when additional checks were removed from the function. Remove after next LTE.
if (func_num_args() !== 1) throw new Exception('Invalid number of parameters passed to SubEditorsDAO::deleteByUserId!');
```
...but the likelihood of this being encountered are close enough to 0 that I'm OK to leave it out if you'd prefer.

(Trivia: [this PHP-level proposal](https://wiki.php.net/rfc/strict_argcount) to warn about excess parameters was proposed but utterly defeated in the vote.)

Reply:
Funny, I guess I just assumed that PHP threw an exception in such cases! I've added in the exception. :+1: